### PR TITLE
Add basic platform support for Android.

### PR DESCRIPTION
**Pull request by grynspan**
This PR seeks out all the nooks and crannies where we have platform-specific code or logic and adds Android. In most cases, it's as simple as changing `os(Linux)` to `os(Linux) || os(Android)` but there are a few spots where they diverge. The PR should be _mostly_ self-explanatory.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

Co-authored-by: Saleem Abdulrasool <compnerd@compnerd.org>

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci test

**Review by compnerd — APPROVED** [review]
Thanks! Should see what the requirements for backtrace are I suppose.

**Comment by grynspan** [comment]
Would be great if you can confirm this builds and (ideally) runs before I merge it. I am not set up to verify it on Android at this time.

**Comment by grynspan** [comment]
@finagolfin We could use your input as well!

**Comment by finagolfin** [comment]
I've never built this repo for Android, in part because it didn't have any snapshot tags till a couple days ago, so don't know anything about it. If and when I do build it, I'll chime in then.

**Comment by grynspan** [comment]
We'll go ahead and move forward with this PR—if you run into issues, please let us know and we can open subsequent PRs to resolve them.

(Waiting for core team review as well.)

**Comment by grynspan** [comment]
@compnerd Note if you need this fix in Swift 6.0.1, please clone the PR and nominate it using the procedure outlined [here](https://forums.swift.org/t/swift-6-0-release-process/70220).

**Pull request merged**
